### PR TITLE
Fixed cursor not being aligned after moving up and down through tabs.

### PR DESCRIPTION
**Pull request by Isogash**
This solution is okay for now, should also start to consider the case of soft wrap maybe? Should this be handled in a core Lua extension later?

### CODE REVIEW

File: src/input.c [hunk 1]
```diff
@@ -97,22 +98,31 @@ void editorMoveCursor(int key) {
...
       } else if (E.cy < E.numrows) {
         E.cy++;
-        E.cx = 0;
+        E.rx = 0;
       }
       break;
     case ARROW_UP:
       if (E.cy != 0) {
         E.cy--;
+        goto vertical;
```
Comment (PIKMYN) [inline src/input.c]: hello

Comment (Isogash) [reply src/input.c]: yes

File: src/input.c [hunk 2]
```diff
@@ -97,22 +98,31 @@ void editorMoveCursor(int key) {
...
       }
       break;
     case ARROW_DOWN:
       if (E.cy < E.numrows) {
         E.cy++;
+        goto vertical;
```
Comment (PIKMYN) [inline src/input.c]: yuck

**Comment by Isogash** [comment]
Please merge

**Comment by Isogash** [comment]
Fixes #9

**Pull request merged**